commit 487fc5c20e0a44082af4db36997946a8ab965f21
Author: Anil Madhavapeddy <anil@recoil.org>
Date:   Mon Jan 20 17:50:20 2014 +0000

    Fix compilation under OpenBSD 5.4
    
    Closes #33

diff --git a/lib/unix_stubs.c b/lib/unix_stubs.c
index 9565da1..560d034 100644
--- a/lib/unix_stubs.c
+++ b/lib/unix_stubs.c
@@ -1562,7 +1562,7 @@ CAMLprim value core_unix_mcast_modify (value v_action,
       }
 
       if (Is_block(v_source_opt)) {
-#if defined(__APPLE__)
+#if defined(__APPLE__) || defined(__OpenBSD__)
         caml_failwith("core_unix_mcast_modify: ~source is not supported on MacOS");
 #else
         struct ip_mreq_source mreq_source;
